src/test/1_HarmonicFunctionsTests.js

```diff
@@ -0,0 +1,164 @@
+}
+harmonicFunctionsTestSuite.addTest(new UnitTest.UnitTest(sikorski_zzip_ex1, "HarmonicFunctions Major test 5 - from sikorski_zzip_ex1"));
+
+
+var sikorski_zzip_ex3 = () => {
+    var input = get_ex_from_file("\\examples\\1_HarmonicFuntions\\major\\sikorski_zzip_ex3.txt");
+    var ex = Parser.parse(input);
+    var solver = new Solver.Solver(ex);
+    var solution = solver.solve();
+
+    return UnitTest.assertDefined(solution.chords[solution.chords.length - 1].sopranoNote);
+}
+harmonicFunctionsTestSuite.addTest(new UnitTest.UnitTest(sikorski_zzip_ex3, "HarmonicFunctions Major test 6 - from sikorski_zzip_ex3"));
+
+
+var sikorski_zzip_ex53 = () => {
+    var input = get_ex_from_file("\\examples\\1_HarmonicFuntions\\major\\sikorski_zzip_ex53.txt");
+    var ex = Parser.parse(input);
+    var solver = new Solver.Solver(ex);
+    var solution = solver.solve();
+
+    return UnitTest.assertDefined(solution.chords[solution.chords.length - 1].sopranoNote);
+}
+harmonicFunctionsTestSuite.addTest(new UnitTest.UnitTest(sikorski_zzip_ex53, "HarmonicFunctions Major test 7 - from sikorski_zzip_ex53"));
+
+
+var sikorski_zzip_ex54 = () => {
+    var input = get_ex_from_file("\\examples\\1_HarmonicFuntions\\major\\sikorski_zzip_ex54.txt");
+    var ex = Parser.parse(input);
+    var solver = new Solver.Solver(ex);
+    var solution = solver.solve();
+
+    return UnitTest.assertDefined(solution.chords[solution.chords.length - 1].sopranoNote);
+}
+harmonicFunctionsTestSuite.addTest(new UnitTest.UnitTest(sikorski_zzip_ex54, "HarmonicFunctions Major test 8 - from sikorski_zzip_ex54"));
+
+
+var sikorski_zzip_ex77 = () => {
+    var input = get_ex_from_file("\\examples\\1_HarmonicFuntions\\major\\sikorski_zzip_ex77.txt");
+    var ex = Parser.parse(input);
+    var solver = new Solver.Solver(ex);
+    var solution = solver.solve();
+
+    return UnitTest.assertDefined(solution.chords[solution.chords.length - 1].sopranoNote);
+}
+harmonicFunctionsTestSuite.addTest(new UnitTest.UnitTest(sikorski_zzip_ex77, "HarmonicFunctions Major test 9 - from sikorski_zzip_ex77"));
+
+
+// MINOR
+var targosz_p61_ex17 = () => {
+    var input = get_ex_from_file("\\examples\\1_HarmonicFuntions\\minor\\targosz_p61_ex17.txt");
+    var ex = Parser.parse(input);
+    var solver = new Solver.Solver(ex);
+    var solution = solver.solve();
+
+    return UnitTest.assertDefined(solution.chords[solution.chords.length - 1].sopranoNote);
+}
+harmonicFunctionsTestSuite.addTest(new UnitTest.UnitTest(targosz_p61_ex17, "HarmonicFunctions Minor test 1 - from targosz_p61_ex17"));
+
+
+var targosz_p61_ex18 = () => {
+    var input = get_ex_from_file("\\examples\\1_HarmonicFuntions\\minor\\targosz_p61_ex18.txt");
+    var ex = Parser.parse(input);
+    var solver = new Solver.Solver(ex);
+    var solution = solver.solve();
+
+    return UnitTest.assertDefined(solution.chords[solution.chords.length - 1].sopranoNote);
```

Comment:
Co sie dzieje w przypadku, gdyby solution nie znalazło i by były te najnizsze nuty wyplute?

Reply:
To jest kwestia reagowania MuseScore na undefined na wejściu, dlatego tutaj ta asercja jest poprawna